github/ci/prow-deploy/kustom/overlays/workloads-production/resources/place-holder.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: sriov-place-holder
+  labels:
+    name: sriov-place-holder
+spec:
+  selector:
+    matchLabels:
+      name: sriov-place-holder
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: sriov-place-holder
+        sriov-pod: "true"
+    spec:
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: sriov-pod
+                operator: In
+                values:
+                - "true"
+            topologyKey: kubernetes.io/hostname
+      nodeSelector:
+        hardwareSupport: sriov-nic
+      terminationGracePeriodSeconds: 1
+      priorityClassName: sriov-place-holder
+      tolerations:
+      - operator: Exists
+        effect: NoSchedule
+      containers:
+      - name: sriov-place-holder
+        image: k8s.gcr.io/busybox
+        command: [ "/bin/sh", "-c", "while true; do sleep 86400; done" ]
+        resources:
+          requests:
+            memory: 29Gi
```

Comment:
This will consume memory for jobs that could be ready to run when no SRIOV jobs are ready to run. Let's have a quick look at how many SRIOV jobs we see in our CI and verify this compromise is relevant.

Reply:
Just spoke with @brianmcarey and @EdDev
we have a buffer so we are safe to reserve this chunk.

Moreover we can't reduce the memory to 18Gi because those numbers are mainly in order to control how many run on the same time, not for how many memory it actually need (it needs 15.6 according the recent check with Brian).

Personally I find killing a random pod problematic, we will need to debug again and again why our jobs failed, even if they will be rescheduled.

@enp0s3 @dhiller FYI

Reply:
> This will consume memory for jobs that could be ready to run when no SRIOV jobs are ready to run. Let's have a quick look at how many SRIOV jobs we see in our CI and verify this compromise is relevant.

What exactly is consumed? The requested values are just plain resources, it does not allocate any memory.

The SR-IOV machines have been introduced to serve SR-IOV workloads, the fact that other workloads run on them is nice, but it should not come on the cost of the SR-IOV ones.

At the moment, SR-IOV jobs are delayed due to non SR-IOV jobs running on these machine. I would like this to be optimized and avoided.

If the CI maintainers prefer to preempt running jobs instead of preserving a slot for the SR-IOV jobs, I do not have a strong objection. I think it is less desired, but I prefer prioritizing SR-IOV as the first priority.

Reply:
>Moreover we can't reduce the memory to 18Gi because those numbers are mainly in order to control how many run on the same time, not for how many memory it actually need (it needs 15.6Gi according the recent check with Brian).

Seems like a hack that should be replaced by a proper mechanism.

>What exactly is consumed? The requested values are just plain resources, it does not allocate any memory.

Kubernetes scheduler is working based on a request, not actual consumption. Therefore you don't get to run different jobs there.

>The SR-IOV machines have been introduced to serve SR-IOV workloads, the fact that other workloads run on them is nice, but it should not come on the cost of the SR-IOV ones.

Agree but let's explore an optimal solution that also provides the best utilization of the machine as we are seeing the requests to maximize our CI capacity.

>At the moment, SR-IOV jobs are delayed due to non SR-IOV jobs running on these machine. I would like this to be optimized and avoided.

Do you have any examples? It would greatly help us understand why this happen and if a simpler solution couldn't be applied.

>If the CI maintainers prefer to preempt running jobs instead of preserving a slot for the SR-IOV jobs, I do not have a strong objection. I think it is less desired, but I prefer prioritizing SR-IOV as the first priority.

The preemption was just an idea brought up during a discussion of this PR. Further exploring of consequences is needed.
Example: It doesn't make sense to use it if we would preempt every 15 minutes but it does make sense if it would be once per day.

Reply:
/hold
@xpivarc Didn't see your comments.

Reply:
No need to hold for this if others feel it's good for now. I just opened discussion about further steps.

Reply:
> Seems like a hack that should be replaced by a proper mechanism.

FYI
There is this extended resource method if we find it helpful
(to create other one to control how many run on each machine)
https://github.com/kubevirt/project-infra/blob/main/hack/patch_node.sh

Reply:
Yes, this would be the right path(IMHO).